src/types/mod.ts

```diff
@@ -30,7 +30,7 @@ export class CustomType<Def extends CustomTypeDefinition>
   }
 
   private generateReferenceString() {
-    return `#/types/${this.id}`;
+    return this.id.startsWith("slack#/") ? this.id : `#/types/${this.id}`;
```

Comment:
I think this is fine, but this reminded me that I solved this slightly differently for function reference strings. There I just did a check to see if the id contained a `"#"` - and if it did, assumed it was a full reference and used it, otherwise built the "local" reference string.

```ts
// ensures the function reference is a full path - local functions will only be passing in the function callback id
    this.functionReference = functionReference.includes("#/")
      ? functionReference
      : `${localFnPrefix}${functionReference}`;
```

It may make no difference, but I thought it may be useful if we ever allowed referencing other app's entities at some point, then things would potentially "just work"... or something.

Reply:
Ah yeah good call! Just going to say this out loud — we may have a problem in the future where we auto-register types from other apps, depending on how we do that. If that will look like `type: A0123/types/example`, no issue. If we're actually importing the `ICustomType` from their app (for typeahead purposes maybe), then we'll need to add logic to handle that. Ex.

```ts
// manifest.ts
import { IncidentType } from "deno-slack-app-repository/incident_app.ts";

DefineFunction({
  ...,
  input_parameters: {
    properties: { incident: { type: IncidentType },
    ...,
  }
});

```